### PR TITLE
add org.libretrainsim.Libre-TrainSim

### PR DESCRIPTION
**Pull request by Jean28518**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ x My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). *(Will redirect to libre-trainsim.de in a few days)*
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.libretrainsim.community

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"skip-arches": ["aarch64"]
```
Comment (hfiguiere) [inline flathub.json:2]:
```suggestion
	"only-arches": ["x86_64"]
```

File: org.libretrainsim.community.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+	"app-id": "org.libretrainsim.community",
```
Comment (hfiguiere) [inline org.libretrainsim.community.json:2]: This is a weird app ID: it make us think that the name of the application is "community".

File: org.libretrainsim.community.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	</screenshot>
+  </screenshots>
+  
+  <releases>
+	<release date="2021-08-13" version="Alpha v0.8">
+	  <description>
+		<p>
+		  Improved / Changed
+		  <ul>
+            <li> Passengers have arrived to Libre TrainSim! There exist </li>
```
Comment (hfiguiere) [inline org.libretrainsim.community.metainfo.xml:64]:
```suggestion
            <li> Passengers have arrived to Libre TrainSim! They exist </li>
```

File: org.libretrainsim.community.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            <li> Camera Shaking in Cabin View </li>
+            <li> Track Files got much smaller. </li>
+            <li> Loading time was improved a lot </li>
+            <li> Implemented jTools: self written toolset, which makes code cleaner and functionality more stable </li>
+            <li> Soundsystem got rewritten. Much more cleaner and mor modular </li>
+            <li> Added 32 bit Version vor Linux </li>
+            <li> Libre TrainSim is now accessible for smaller Displays </li>
+            <li> Trains have now a engine function. Before releasing brakes you have to press e to turn on the engines. </li>
+            <li> Sounds are now played when train drives over switches and curves </li>
+            <li> Rails now have a overheadline </li>
```
Comment (hfiguiere) [inline org.libretrainsim.community.metainfo.xml:77]:
```suggestion
            <li> Rails now have a overhead line </li>
```

**Comment by hfiguiere** [comment]
will the appdata and .desktop file be added to the upstream repository and maintained there?

**Comment by Jean28518** [comment]
Thanks for your fast reaction!

> will the appdata and .desktop file be added to the upstream repository and maintained there?

do you mean with appdata Hainfurt.pck and U2-Nuremberg.pck? 
I could host and maintain them in a new github repository. Want to keep them separate to the main code. They are just content.

.desktop file will be maintained in https://github.com/Jean28518/Libre-TrainSim if you whish.

> This is a weird app ID: it make us think that the name of the application is "community".

Should I call it `org.libretrainsim.Libre-TrainSim` ? No problem.

Thank you very much for your grammar corrections!

**Comment by hfiguiere** [comment]
> Thanks for your fast reaction!
> 
> > will the appdata and .desktop file be added to the upstream repository and maintained there?
> 
> do you mean with appdata Hainfurt.pck and U2-Nuremberg.pck?
> I could host and maintain them in a new github repository. Want to keep them separate to the main code. They are just content.

No. I mean the file `org.libretrainsim.community.metainfo.xml`.
 
> .desktop file will be maintained in https://github.com/Jean28518/Libre-TrainSim if you whish.

Neither are flatpak specific just to be clear, but flatpak require them. The .desktop is already being widely used by desktop environments to have the application appear in "menus". 

> > This is a weird app ID: it make us think that the name of the application is "community".
> 
> Should I call it `org.libretrainsim.Libre-TrainSim` ? No problem.

This would be better IMHO.

**Comment by Jean28518** [comment]
Thanks.

So to understand you correctly: I have to upload org.libretrainsim.community.metainfo.xml and org.libretrainsim.community.desktop to https://github.com/Jean28518/Libre-TrainSim ?

**For renaming:** Should i create a new pull request from a new branch which is called `org.libretrainsim.Libre-TrainSim` and we can close this one? Or can we rename everything in here?

Sorry, thats my first flatpak contribution. Thanks for your patience!

**Comment by hfiguiere** [comment]
> Thanks.
> 
> So to understand you correctly: I have to upload org.libretrainsim.community.metainfo.xml and org.libretrainsim.community.desktop to https://github.com/Jean28518/Libre-TrainSim ?

That would be nice if they were indeed maintained upstream, at least for future release.

> 
> **For renaming:** Should i create a new pull request from a new branch which is called `org.libretrainsim.Libre-TrainSim` and we can close this one? Or can we rename everything in here?

Branch name doesn't matter. You can just rename things and update your branch, the pull request will be updated when you push. What matter is the name of the manifest, .destkop and .metainfo.xml.

**Comment by hfiguiere** [comment]
> > Thanks.
> > So to understand you correctly: I have to upload org.libretrainsim.community.metainfo.xml and org.libretrainsim.community.desktop to https://github.com/Jean28518/Libre-TrainSim ?
> 
> That would be nice if they were indeed maintained upstream, at least for future release.

Just to clarify, it's OK to have these here for the time being.

**Comment by Jean28518** [comment]
Okay, thanks!
I renamed everything.

**Comment by hfiguiere** [comment]
bot, build org.libretrainsim.Libre-TrainSim

**Comment by nedrichards** [comment]
This all looks good to me. The only thing that would be ideal would be setting up an extension point for content packs to be added since currently the app says 'add them to /app/bin' which is read only inside the deployed flatpak.

File: org.libretrainsim.Libre-TrainSim.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+				},
+				{
+					"type": "file",
+					"path": "org.libretrainsim.Libre-TrainSim.desktop"
+				}
+			]
+		}
+	],
+	"finish-args": [
+		"--socket=x11",
```
Comment (barthalion) [inline org.libretrainsim.Libre-TrainSim.json:52]:
```suggestion
		"--socket=fallback-x11",
```

**Comment by barthalion** [comment]
Alternatively, re-using `~/.local/share/godot/app_userdata/libre-trainsim` or `$XDG_DATA_HOME` for new content packs would be also good.

**Comment by nedrichards** [comment]
bot, build org.libretrainsim.Libre-TrainSim

**Pull request closed without merging**